src/train_tokeniser.py

```diff
@@ -0,0 +1,142 @@
+        normaliser_list.append(normalizers.Lowercase())
+    normaliser = normalizers.Sequence(normaliser_list)  # noqa
+    tokeniser.normalizer = normaliser
+
+    # Shorthand for whether a prefix whitespace should be added to words
+    pre_ws = config.add_prefix_space
+
+    # Initialise the pre-tokeniser and add it to the tokeniser
+    pre_tok_list = list()
+    if config.byte_level:
+        pre_tok_list.append(pre_tokenizers.ByteLevel(add_prefix_space=pre_ws))
+    if config.sentence_piece:
+        pre_tok_list.append(pre_tokenizers.Metaspace(add_prefix_space=pre_ws))
+    if not config.byte_level and not config.sentence_piece:
+        pre_tok_list.append(pre_tokenizers.Whitespace())
+    pre_tokeniser = pre_tokenizers.Sequence(pre_tok_list)
+    tokeniser.pre_tokenizer = pre_tokeniser
+
+    # Initialise the post-processor
+    if config.add_sep_and_cls_tokens:
+        params = dict(cls=(config.bos_token, 1),
+                      sep=(config.eos_token, 2),
+                      trim_offsets=True,
+                      add_prefix_space=pre_ws)
+        tokeniser.post_processor = processors.RobertaProcessing(**params)
+    elif config.byte_level:
+        tokeniser.post_processor = processors.ByteLevel(trim_offsets=True)
+
+    # Initialise the decoder
+    if config.tokeniser_type == 'bpe':
+        tokeniser.decoder = decoders.BPEDecoder()
+    elif config.tokeniser_type == 'wordpiece':
+        tokeniser.decoder = decoders.WordPiece(prefix='##', cleanup=True)
+    elif config.sentence_piece:
+        tokeniser.decoder = decoders.Metaspace(add_prefix_space=pre_ws)
+    elif config.byte_level:
+        tokeniser.decoder = decoders.ByteLevel()
+
+    # Enable truncation
+    if config.truncation:
+        tokeniser.enable_truncation(max_length=config.max_length)
+
+    # Enable padding
+    if config.padding:
+        tokeniser.enable_padding(pad_id=0,
+                                 pad_type_id=0,
+                                 pad_token=config.pad_token)
+
+    # Initialise the trainer
+    if config.tokeniser_type == 'bpe':
+        trainer = trainers.BpeTrainer(vocab_size=config.vocab_size,
+                                      show_progress=show_progress,
+                                      special_tokens=special_tokens)
+    elif config.tokeniser_type == 'wordpiece':
+        trainer = trainers.WordPieceTrainer(vocab_size=config.vocab_size,
+                                            show_progress=show_progress,
+                                            special_tokens=special_tokens)
+    elif config.tokeniser_type == 'unigram':
+        trainer = trainers.UnigramTrainer(vocab_size=config.vocab_size,
+                                          special_tokens=special_tokens,
+                                          show_progress=show_progress,
+                                          unk_token='<unk>')
+
+    # Train the tokeniser
+    tokeniser.train_from_iterator(iterator=dataset['text'],
```

Comment:
Does this work for streaming datasets:

```
ds = load_dataset("DDSC/angry-tweets", streaming = True)
ds = ds["train"]
ds["text"]
# TypeError: 'IterableDataset' object is not subscriptable
```

One option would be to just take in an iterator?

Reply:
then you could simply input:
```
(e["text"] for e in iterable_dataset)
```

Reply:
Good catch! I've implemented the iterator now, and included a unit test for streamed datasets.